### ISSUE TITLE
Question about  syncBN

### ISSUE THREAD

**Issue by Gaondong** [tensorpack/tensorpack#1516]
Thanks for your great works.

1. I want use the syncBN in my tensorflow project and I saw the code in your project, I read all the codes of BatchNorm in `tensorpack/models/batch_norm.py`, but still don't understand the meaning of `vh=ret.variables=VariableHolder` in the below codes.
2. Did the `ret` really return the `tf.identity` which same as the output of `tf.nn.batch_normalization`?
3. Have you dealt with the back propagation problem caused by BN [synchronization?](http://hangzhang.org/mxnet-zh-blog/syncbn)?(http://hangzhang.org/mxnet-zh-blog/syncbn)

`if new_shape is not None:
    # Using fused_batch_norm(is_training=False) is actually slightly faster,
    # but hopefully this call will be JITed in the future.
    xn = tf.nn.batch_normalization(
        inputs, batch_mean, batch_var,
        tf.reshape(beta, new_shape),
        tf.reshape(gamma, new_shape), epsilon)
else:
    xn = tf.nn.batch_normalization(
        inputs, batch_mean, batch_var,
        beta, gamma, epsilon)
if virtual_batch_size is not None:
    xn = tf.reshape(xn, orig_shape)

if do_ema_update:
    ret = internal_update_bn_ema(
        xn, batch_mean_vec, batch_var_vec, moving_mean, moving_var, momentum)
else:
    ret = tf.identity(xn, name='output')

vh = ret.variables = VariableHolder(
    moving_mean=moving_mean,
    mean=moving_mean,  # for backward-compatibility
    moving_variance=moving_var,
    variance=moving_var)  # for backward-compatibility
if scale:
    vh.gamma = gamma
if center:
    vh.beta = beta
return ret`

**Comment by ppwwyyxx** [comment]
1. it's not needed and does not affect computation
2. I can't understand the question - `ret` is in the code and you can see what it is
3. I can't understand what is "the back propagation problem" - there is no problem as far as I can see.

**Comment by Gaondong** [comment]
> 1. it's not needed and does not affect computation
> 2. I can't understand the question - `ret` is in the code and you can see what it is
> 3. I can't understand what is "the back propagation problem" - there is no problem as far as I can see.
I saw syncBN method in the link http://hangzhang.org/mxnet-zh-blog/syncbn, it said  in back propagation should  synchronization mean x and square x.

**Comment by ppwwyyxx** [comment]
TensorFlow does back propagation automatically.

**Comment by Gaondong** [comment]
> TensorFlow does back propagation automatically.

Thanks for your reply, I will try it.
